src/main/java/org/codevillage/PutShapesChainEnd.java

```diff
@@ -0,0 +1,11 @@
+package org.codevillage;
+
+public class PutShapesChainEnd extends ShapeChainEnd{
+    @Override
+    public void construct() {
+        CanvasData canvasData = CanvasData.getInstance().getShapes();
+        ShapePositioningData shapePositioningData = ShapePositioningData.getInstance().getShapes();
```

Comment:
Same with this. I think the types of these should be `List<Shape>` instead of CanvasData and ShapePositioningData, right?

Reply:
Yeah that's right, i'll make those changes